IPs/RSKIP241.md

```diff
@@ -0,0 +1,120 @@
+
+The input selector algorithm is modified so that a reasonable attempt is made to choose inputs that can cover potentially at least four times the computed transaction fees. 
+
+These methods create new release transactions, but the Bridge does not store store them in its storage. The new transaction hashes are not stored in *releasedTransactionIds*.
+
+### Method feeBump()
+
+* If *feeToAdd* >= *T.change*, then then a new transaction *T'* is built with the same inputs, the same outputs, but with *T'.change = T.change-feeToAdd*. This moves fees from the change output to the transaction fee. If *T'.change* is low the dust limit, the output is removed.
+* Otherwise, REVERT
+
+Note that because of the change in the input selector algorithm, the case that  *feeToAdd* >= *T.change* will be the most common. This method allows to bump a transaction in most cases, but can only provide high assurance up to a 2X bump of the fees.
+
+### Method feeBumpWithInput()
+
+A new transaction *T'* is built having the inputs and outputs of *T*, but with an additional input appended, specified by newInput. The user is responsible to check that the input with amount is enough to bump the fees. This is not checked by the Bridge. The input given must already be signed.
+
+The full amount of the input will be consumed to pay fees.
+
+ The user is responsible to check that the input with amount is enough to bump the fees. This is not checked by the Bridge.
+
+### Method feeBumpWithCPFP()
+
+This method performs all checks and transaction changes similar to feeBump(). Then the new transaction *T'* is appended an additional output containing a transfer of *pot*  bitcoins paying to the address 1CPFPxxxSv1jWf6wCyihRWhKyYFkKX3bnf. The private key for this address is 5KFhJj5HeGh81EuzuRWo6SigTN5cQnFSTmeSHsWuwG7AN5pze9C. Since the inputs have not changed, the fee per byte of *T'* will be lower than the fee per byte of *T* unless the caller also adds fees by transferring bitcoins in the call. Because *T'* must replace *T* in the memory pool, an additional fee will be required. The may need use a Child-Pay-For-Parent later, consuming the CPFPxxx output to push this transaction into a block. This method allows to confirm a transaction if the user suspect there is a risk that the transaction fees increase an order of magnitude in a short time.
+
+## Rationale
+
+The following topics has been considered in this proposal.
+
+### More protections against DoS
+
+The high gas cost of these methods is to prevent users abusing of the powpeg resources to sign transaction. To prevent abuse even more Bridge could allow only fee bumps that add at least 30% more fees that the previous fees used. Therefore each attempt becomes more and more expensive. This can be done by storing the last fee used in *releasedTransactionIds*.
```

Comment:
```suggestion
The high gas cost of these methods is to prevent users abusing of the powpeg resources to sign transactions. To prevent abuse even more, the Bridge could allow only fee bumps that add at least 30% more fees that the previous fees used. Therefore, each attempt becomes more and more expensive. This can be done by storing the last fee used in *releasedTransactionIds*.
```